src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.Android.cs

```diff
@@ -421,24 +423,24 @@ private void UpdateTextBoxViewReadOnly()
 				return;
 			}
 
-			_textBoxView.Focusable = IsTabStop;
-			_textBoxView.FocusableInTouchMode = IsTabStop;
-			_textBoxView.Clickable = !IsNativeReadOnly;
-			_textBoxView.LongClickable = !IsTabStop;
-			_textBoxView.SetCursorVisible(!IsNativeReadOnly);
-
-			if (IsNativeReadOnly)
-			{
-				_listener = _textBoxView.KeyListener;
-				_textBoxView.KeyListener = null;
-			}
-			else
+			if (IsNativeViewReadOnly)
 			{
-				if (_listener != null)
+				if (_textBoxView.KeyListener is not null)
 				{
-					_textBoxView.KeyListener = _listener;
+					_listener = _textBoxView.KeyListener;
+					_textBoxView.KeyListener = null;
 				}
 			}
+			else if (_listener is not null)
+			{
+				_textBoxView.KeyListener = _listener;
+			}
+
+			_textBoxView.Focusable = IsTabStop;
+			_textBoxView.FocusableInTouchMode = IsTabStop;
+			_textBoxView.Clickable = IsTabStop;
+			_textBoxView.LongClickable = IsTabStop;
```

Comment:
This is the opposite of the original code, is this the fix?

Reply:
Yes, actually the read-only part is handled by the KeyListener. The IsTabStop should decide the interactivity, I have tested all 4  different configurations and it behaves as in WinUI (at least on my side)

Reply:
Will verify this

Reply:
Finally found the solution!